packages/ses/src/module-load.js

```diff
@@ -297,6 +323,64 @@ const memoizedLoadWithErrorAnnotation = async (
   return moduleLoading;
 };
 
+function asyncOverseer() {
+  /** @type {Set<Promise<undefined>>} */
+  const pendingJobs = new Set();
+  /** @type {Array<Error>} */
+  const errors = [];
+
+  /**
+   * Enqueues a job that starts immediately but won't be awaited until drainQueue is called.
+   *
+   * @template {any[]} T
+   * @param {(...args: T)=>Promise<*>} func
+   * @param {T} args
+   */
+  const enqueueJob = (func, args) => {
+    setAdd(
+      pendingJobs,
+      promiseThen(func(...args), noop, error => {
+        arrayPush(errors, error);
+      }),
+    );
+  };
+  /**
+   * Sequentially awaits pending jobs and returns an array of errors
+   *
+   * @returns {Promise<Array<Error>>}
+   */
+  const drainQueue = async () => {
+    for (const job of pendingJobs) {
+      // eslint-disable-next-line no-await-in-loop
+      await job;
+    }
+    return errors;
+  };
+  return { enqueueJob, drainQueue };
+}
+
+/**
+ * @param {object} options
+ * @param {Array<Error>} options.errors
+ * @param {string} options.errorPrefix
+ */
+function throwAggregateError({ errors, errorPrefix }) {
+  // Throw an aggregate error if there were any errors.
+  if (errors.length > 0) {
+    const verbose =
+      getenv('COMPARTMENT_LOAD_ERRORS', '', ['verbose']) === 'verbose';
+    throw TypeError(
+      `${errorPrefix} (${errors.length} underlying failures: ${arrayJoin(
+        arrayMap(errors, error => error.message + (verbose ? error.stack : '')),
+        ', ',
+      )}`,
+    );
+  }
+}
```

Comment:
We may now feature detect `AggregateError` and use it. It is admitted into permits now.

Reply:
I did put it in and it looks great. 
It affects so many tests, I want it to be a separate PR and we need to decide whether it makes for a breaking change.